commands/file_uploader.go

```diff
@@ -62,7 +62,9 @@ func (f *FileUploader) UploadFiles() ([]*models.FileMetadata, ExecutionStatus) {
 		if !isFileAlreadyUploaded(file, fileInfo, uploadedMtaFiles, &alreadyUploadedFiles) {
 			// If not, add it to the list of uploaded files
 			fileToUpload, err := os.Open(file)
-			defer fileToUpload.Close()
+			if fileToUpload != nil {
```

Comment:
This nil check is redundant because fileToUpload will be nil only if an error occurs. Even if an error occurs because of the defer statement Close() will be executed after the surrounding function returns.

Reply:
You can just move the defer after the if err != nil check on line 68.  os.Open would either return an error or the file handle, but not both.

Reply:
Done